plugins/org.eclipse.n4js.cli/src/org/eclipse/n4js/cli/compiler/SystemExitRedirecter.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.cli.compiler;
+
+import java.security.Permission;
+
+/**
+ * Catches calls to System.exit() and throws an {@link SystemExitException} instead.
+ */
+public class SystemExitRedirecter {
```

Comment:
Why did you introduce this?

It's a cool idea, but doesn't this make the code less readable / surprising? Why not just use the code with the desired behavior instead of `System.exit()`? Also, not sure about using security managers for ordinary application behavior.

Reply:
Also only for debugging. I removed it again.